lesson_02/load_sales_raw_job1/dal/storage_api_to_disk.py

```diff
@@ -0,0 +1,29 @@
+import json
+import os
+from typing import List, Dict, Any
+
+from load_sales_raw_job1.dal.exceptions import SalesDalStorageSaveException
+from load_sales_raw_job1.dal.storage_api import StorageDalInterface
+from logs_handling.log_item import LogItemInterface
+from support_tools.file_tools import guarantee_folder_exists, logical_to_physical_file_path
+from logs_handling.log_formatter import LogFormatter
+
+
+class StorageDalDisk(StorageDalInterface):
+    def __init__(self, root_path: str):
+        super(StorageDalDisk, self).__init__()
+        self.__root_path = os.path.normpath(os.path.normcase(root_path))
+
+    def save(self, log: LogItemInterface, json_content: List[Dict[str, Any]],
+             logical_path: str, file_name: str) -> None:
+        full_file_path = logical_to_physical_file_path(self.__root_path, logical_path, file_name)
+        full_folder_path = os.path.dirname(full_file_path)
+        guarantee_folder_exists(full_folder_path)
+        try:
+            if os.path.isfile(full_file_path):
```

Comment:
you should clean the whole directory instead of deleting file by file. Because you don't know the quantity of files you get on each the same request (ideally, it should be the same number each time because GET request should be idempotent, but our world is not perfect)